shell/platform/embedder/embedder.cc

```diff
@@ -56,76 +56,68 @@ bool IsRendererValid(const FlutterRendererConfig* config) {
   return true;
 }
 
-struct _FlutterPlatformMessageResponseHandle {
-  fml::RefPtr<blink::PlatformMessage> message;
-};
-
-FlutterResult FlutterEngineRun(size_t version,
-                               const FlutterRendererConfig* config,
-                               const FlutterProjectArgs* args,
-                               void* user_data,
-                               FlutterEngine* engine_out) {
-  // Step 0: Figure out arguments for shell creation.
-  if (version != FLUTTER_ENGINE_VERSION) {
-    return kInvalidLibraryVersion;
+static bool IsSoftwareRendererConfigValid(const FlutterRendererConfig* config) {
+  if (config->type != kSoftware) {
+    return false;
   }
 
-  if (engine_out == nullptr) {
-    return kInvalidArguments;
+  const FlutterSoftwareRendererConfig* software_config = &config->software;
+
+  if (SAFE_ACCESS(software_config, surface_present_callback, nullptr) ==
+      nullptr) {
+    return false;
   }
 
-  if (args == nullptr) {
-    return kInvalidArguments;
+  return true;
+}
+
+static bool IsRendererValid(const FlutterRendererConfig* config) {
+  if (config == nullptr || config->type != kOpenGL) {
```

Comment:
Given the checks below, why the `config->type != kOpenGL`?

Reply:
I wanted the method to be self contained.